### DISCUSSION TITLE
Example using `references` in a field schema

### DISCUSSION THREAD

**Discussion by meadoch1** [sensedeep/dynamodb-onetable#435] [Q&A] [answered]
I have a schema where one model includes fields that are the primary keys of two other models.  I see the `references` element in a field definition that says it can contain `model:index:attribute=source_attribute` to establish a reference between the two; however, I do not find any concrete example of using this property in the samples, tests, discussions, or source code.  Would it be possible to get a small example to show how this works?

**Comment by mobsense** [comment]
References are used by 3rd party tools like SenseDeep to be able to follow such references as hot links.

In SenseDeep, such references appear as blue underline hot links that you can click on to follow to the referenced item. Very convenient!

**Reply by meadoch1** [reply]
So it's not really intended to be used in development of an application then I guess.  Thanks.

**Comment by dev-embedthis** [comment, marked as answer]
Yes. But you and some do, use it in code to understand the "owner" of an item.

For example:

```
        Device: {
            pk:             { type: 'string', value: 'd#${id}' },
            sk:             { type: 'string', value: 'd#' },

            gs1pk:          { type: 'string', value: 'd#${productId}' },
            gs1sk:          { type: 'string', value: 'd#${id}' },

            id:             { type: 'string', required: true, encode: ['pk', '#', 1] },
            cloudId:        { type: 'string', map: 'cl', reference: 'Cloud:primary:id=cloudId'  },
            productId:      { type: 'string', map: 'pr', reference: 'Product:primary:id=productId'  },
            model:          {type: 'string', map: 'mo'},
            test:           { type: 'boolean', map: 'te' },
            created:        { type: 'date' },
        },
```

This tells us that the device.cloudId can be used as the Cloud.id to retrieve the associated cloud for the device.
